lib/pagy/frontend.rb

```diff
@@ -34,7 +34,7 @@ def pagy_info(pagy)
 
     # this works with all Rack-based frameworks (Sinatra, Padrino, Rails, ...)
     def pagy_url_for(n)
-      url    = File.join(request.script_name.to_s, request.path_info)
+      url    = "#{request.script_name}/#{request.path_info}"
```

Comment:
Actually, this is not even needed. According to Rack specification:

> The `SCRIPT_NAME`, if non-empty, must start with `/`
> The `PATH_INFO`, if non-empty, must start with `/`
> One of `SCRIPT_NAME` or `PATH_INFO` must be set. `PATH_INFO` should be `/` if `SCRIPT_NAME` is empty. `SCRIPT_NAME` never should be `/`, but instead be empty.

And in fact `Rack` provides a handy `#path` method for exactly this purpose:

https://github.com/rack/rack/blob/42e48013dd1b6dbda990dfa3851856c199b0b1f9/lib/rack/request.rb#L406-L408

Reply:
So why don't we use just `url = request.path`?

Reply:
That would be my recommendation as well. I’ll update the PR, perhaps add a few tests.

Reply:
Thanks